wagtail/api/v2/tests/test_pages.py

```diff
@@ -683,6 +683,15 @@ def test_filtering_boolean_validation(self):
             },
         )
 
+    def test_slug_field_containing_null_bytes_gives_error(self):
+        input_with_null_byte = "1\x00����%2527%2522"
```

Comment:
Nit: Could this string use escape codes instead of literals? Eg `\0` for the null byte.

Reply:
Yep, no problem :), see [173101b](https://github.com/wagtail/wagtail/pull/12072/commits/173101b4804c1098c03bd5f2c3fe9ebd625e82bc) and [dee3772](https://github.com/wagtail/wagtail/pull/12072/commits/dee377299c73172dfb2de68fa03760c7fcb3cd63).